plugins/sentinelone/komand_sentinelone/util/api.py

```diff
@@ -35,8 +77,8 @@ def search_agents(
                     endpoint = f"{self.url}web/api/v{api_version}/agents?{search}={agent}"
                     output = requests.get(endpoint, headers=self.token_header)
 
-                    if output.status_code == 200 and output.json()["pagination"]["totalItems"] >= 1:
-                        results.append(output.json()["data"][0])
+                    if output.status_code == 200 and output.json().get("pagination", {}).get("totalItems", 0) >= 1:
+                        results.append(output.json().get("data", [])[0])
```

Comment:
This can still fail with an IndexError - specifically `output.json().get("data", [])[0]` where you try to access the 0 index of the default value list (if `data` is not present). It may be better to split this into multiple lines - check if the 0 index is present and then if so append the result, otherwise do nothing.